libopenage/rng/rng.cpp

```diff
@@ -110,18 +146,18 @@ void RNG::seed(const void *dat, size_t count) {
 
 	constexpr uint64_t default_seed = 0x0123456789abcdef;
 	constexpr size_t uint64_s = sizeof(uint64_t);
-	const char *data = static_cast<const char *>(dat);
-
-	if (count > uint64_s) {
-		init_hash(state, from_cptr(data));
-		hash_bytes(state, data, count - uint64_s);
-	}
-	else {
-		uint64_t init_seed = 0;
-		memcpy(&init_seed, data, count);
-		init_hash(state, init_seed);
-	}
-
+	const uint8_t *data = static_cast<const uint8_t *>(dat);
+
+	// Previously, for state[1] we hashed (data ^ state[0])
+	// I think that now the easiest way to do that is xoring the key
+	uint64_t key[2];
+	memcpy(key, siphash_key, uint64_s * 2);
```

Comment:
As far I understand that the options we have are (correct me if I'm wrong :P):

- Use a hardcoded key, but it's going to be visible to everyone who reads the code
- Generate a key from something like current time, but players will not be able to share seeds
- Generate a key from the seed, but I think that it defeats the purpose of having a key

I think that any of those option are okay